### PR TITLE
refactor(parser): convert functional tests to unit tests

### LINKED ISSUE #2631: Tech debt: convert parser functional tests to unit test and modify the hardcoded values
### Why is this needed?

We are about to create a RFC for Pydantic 2.0 and before that, we need to fix small things. This is a great opportunity to get this tech debt done. 

Also, we need to refactor these tests to avoid hardcoded values to make them more flexible and maintainable.

### Which area does this relate to?

Parser

### Suggestion

_No response_

### Acknowledgment

- [X] This request meets [Powertools for AWS Lambda (Python) Tenets](https://docs.powertools.aws.dev/lambda/python/latest/#tenets)
- [ ] Should this be considered in other Powertools for AWS Lambda languages? i.e. [Java](https://github.com/aws-powertools/powertools-lambda-java/), [TypeScript](https://github.com/aws-powertools/powertools-lambda-typescript/), and [.NET](https://github.com/aws-powertools/powertools-lambda-dotnet/)

### PR DESCRIPTION
**Pull request by leandrodamascena**
**Issue number:** #2631 

### Changes

This pull request aims to enhance the test structure and organization within the parser module. 

We've changed functional tests to unit tests, which makes more sense for the type of testing we're doing. Just to explain a little more, these tests specifically test whether given an event, Powertools can parser that and validate the schema.

### User experience

There is no change in the user experience.
## Checklist

If your change doesn't seem to apply, please leave them unchecked.

* [x] [Meet tenets criteria](https://docs.powertools.aws.dev/lambda/python/#tenets)
* [x] I have performed a self-review of this change
* [x] Changes have been tested
* [ ] Changes are documented
* [x] PR title follows [conventional commit semantics](https://github.com/aws-powertools/powertools-lambda-python/blob/develop/.github/semantic.yml)

<details>
<summary>Is this a breaking change?</summary>

**RFC issue number**:

Checklist:

* [ ] Migration process documented
* [ ] Implement warnings (if it can live side by side)

</details>

## Acknowledgment

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

**Disclaimer**: We value your time and bandwidth. As such, any pull requests created on non-triaged issues might not be successful.

### CODE REVIEW

**Comment by heitorlessa** [comment]
looking 👀

File: tests/unit/parser/test_alb.py [hunk 1]
```diff
@@ -0,0 +1,30 @@
+import pytest
+
+from aws_lambda_powertools.utilities.parser import ValidationError, event_parser
+from aws_lambda_powertools.utilities.parser.models import AlbModel
+from aws_lambda_powertools.utilities.typing import LambdaContext
+from tests.functional.utils import load_event
+
+
+@event_parser(model=AlbModel)
```
Comment (heitorlessa) [inline tests/unit/parser/test_alb.py]: move `event_parser` to a functional test, so you can focus on the Model as the unit to be tested.

For envelopes, you can use `parse` function and pass both `model=, envelope=` arguments.

Everything else looks great!! Such a relief to have this now -- we're gonna be one step away from mutation testing.

**Review by heitorlessa — CHANGES_REQUESTED** [review]
Note to remove functional bits into a functional test (event_parser decorator)

File: tests/events/kinesisSingeEvent.json
Comment (jplock) [inline tests/events/kinesisSingeEvent.json:1]: should this be called `kinesisSingleEvent.json`?

Comment (leandrodamascena) [reply tests/events/kinesisSingeEvent.json:1]: Agree! We may need a better name :)

Let me change that by addressing Heitor's feedback.

**Review by heitorlessa — APPROVED** [review]
SUPERB 🎉 🚀 🚢 🚢

**Pull request merged**